app/(dashboard)/dashboard/interviews/_actions/export.ts

```diff
@@ -0,0 +1,147 @@
+'use server';
+
+import FileExportManager from '~/lib/network-exporters/FileExportManager';
+import { api } from '~/trpc/server';
+import { getServerSession } from '~/utils/auth';
+import { formatExportableSessions, getRemoteProtocolID } from './utils';
+import { trackEvent } from '~/analytics/utils';
+import { type ExportOptions } from '../_components/ExportInterviewsDialog';
+
+type UploadData = {
+  key: string;
+  url: string;
+  name: string;
+  size: number;
+};
+
+type UpdateItems = {
+  statusText: string;
+  progress: number;
+};
+
+type FailResult = {
+  data: null;
+  error: unknown;
+  message: string;
+};
+
+type SuccessResult = {
+  data: UploadData;
+  error: null;
+  message: string;
+};
+
+export const exportSessions = async (
+  interviewIds: { id: string }[],
+  exportOptions?: ExportOptions,
+) => {
+  const session = await getServerSession();
+
+  if (!session) {
+    throw new Error('You must be logged in to export interview sessions!.');
+  }
+
+  const installedProtocols = await api.protocol.get.all.query();
+  const interviewsSessions =
+    await api.interview.get.manyByIds.query(interviewIds);
```

Comment:
Refactor so that you do not need to fetch `installedProtocols` separately, and instead include the protocol for each Session in that tRPC endpoint.

Reply:
Done ✅